src/main/java/com/amihaiemil/docker/Networks.java

```diff
@@ -50,6 +52,31 @@
     Network create(final String name)
         throws IOException, UnexpectedResponseException;
 
+    /**
+     * Create a network with additional parameters.
+     * @param name Name of the network
+     * @param driver Name of the network driver plugin to use
+     * @param ipam Map containing:
+     *  "Driver": Name of the IPAM driver to use
+     *  List of IPAM configuration options:
+     *  "Subnet": CIDR
+     *  "IPRange": CIDR
+     *  "Gateway": IP address
+     *  "AuxAddress": device_name:IP address
+     *  Driver-specific options, specified as a map.     *
+     * @param options Network specific options to be used by the drivers.
+     * @param labels User-defined key/value metadata.
+     * @throws IOException If something goes wrong.
+     * @throws UnexpectedResponseException If the status response is not
+     *  the expected one (200 OK).
+     * @return The created network.
+     * @see <a href="https://docs.docker.com/engine/api/v1.35/#operation/NetworkCreate">Create a network</a>
+     * @checkstyle ParameterNumber (3 lines)
+     */
+    Network create(final String name, final String driver, final Map<String,
+        String> ipam, final Map<String, String> options, final Map<String,
+        String> labels) throws IOException, UnexpectedResponseException;
```

Comment:
@paulodamaso Pay attention, all these parameters should in fact be a JsonObject (see Containers.create(...)). If you look closely, also in the documentation it is a JsonObject (on the right panel, you see it's an http POST with a Json body).

Reply:
@paulodamaso I agree with @amihaiemil. Since name is the only required parameter, please make everything else optional in JsonObject. Just provide another constructor with that optional parameter.

Reply:
@amihaiemil @bkuzmic I disagree. I think that we work too much with JSON instead using real objects. One thing is our objects use JSON to comunicate with the API, another is we use it anywhere just for this convenience. I think that we should make this as decoupled as possible. WDYT?

Reply:
@paulodamaso You want to hide the Json stuff for fear that in the future we might use XML or something else besides Json, but this is never going to happen.

The change would be too big -- not only for input parameters, but also every object in the library is JSON-polymorphic, right? If there will be a new version of the API, with advanced content-negociation, then we'll besically have to write a new library as well :)

Another argument for hiding the JsonObject input stuff is to provide better semantics (the user clearly sees variable names and number of expected params) -- this seems like a good reason, but they are going to have to learn the API anyway (study the docs), in order to understand what those params are doing.

On the other hand, if we have JsonObject: 

* We actually have more flexibility, we don't care if the next version of Docker's API deprecates or invalidates some parameters, the body will still be a JSON. So with this flexibility you have an even more important decoupling: you don't tie yourself to any particular API version.

* The user can make their own json-polimorphic input objects, to give more semantics and reuse format.

This being the case, I still stand with JsonObject inputs :)

Reply:
@amihaiemil Actually I want to hide the JSON stuff because conceptually we shouldn't be tied to these implementations at interface level. I think we should work only with abstract objects. In fact I think that these map parameters could also be transformed in object themselves, but they would be just data bags so I've choose to make them as a map, which I consider something like a "primitive type".
Let's counter your arguments:
> We actually have more flexibility, we don't care if the next version of Docker's API deprecates or > invalidates some parameters, the body will still be a JSON. So with this flexibility you have an 
> even more important decoupling: you don't tie yourself to any particular API version.
I don't see this as a good thing; it would make our implementation mask errors and problems if Docker's API changes its parameters. I think that we should have the failfast approach here and let our lib fail if there is some changes in the supporting services that we interact to.

> javax.json.JsonObject is an interface so the user can make their own json-polimorphic input 
> objects, to give more semantics and reuse format.
This would make our lib receives anything in every method, and I really don't think it's a good idea to keep this level of lack of semantics.

About parametrizing the resource I dont' think it would help in anything, because the implementations would be very different one of another, so in my oppinion I think that we could think about it later.

Reply:
@paulodamaso Still doesn't convince me, sorry. 

> I don't see this as a good thing; it would make our implementation mask errors and problems if Docker's API changes its parameters. 

Not really, because the user sets the API's version in the constructors or ``LocalDocker`` and ``RemoteDocker``. Maybe it's a good idea to remove our implicit assumption of the API's version (we currently provide a default version in the default ctor) -- however, this change we'll make in a future ticket. 

> This would make our lib receives anything in every method,

So what do we care? Of course, the user would be able to pass any kind of ``JsonObject``, but it's obvious that the expected format is described in the API's docs (if it's not obvious, we can specify it in the JavaDocs). And if they still pass garbage Json, then it's their problem. 

One more thing I don't like about explicit params is that we have methods with too many parameters out of which 2 are Strings, 2 are Maps etc - seems very dirty. Another solution would be to use Builders, but in this case we can also keep the JsonObject and provide some input objects ourselves (which would implement ``JsonObjectBuilder`` envelopes).

Nevertheless, if you really want to keep the discussion going, open another ticket and use JsonObject here for now.